p2p/syncer.go

```diff
@@ -536,6 +542,30 @@ func (s *Syncer) syncToPeer(peer gateway.Header) {
 	}
 }
 
+// maintainHealthyPeerSet tries to add peers to the syncer until it has 8.
+func (s *Syncer) maintainHealthyPeerSet() {
+	seen := make(map[string]struct{})
+	for {
+		s.cond.L.Lock()
+		for len(s.peers) >= 8 {
+			// cond is signalled upon disconnecting from a peer
+			s.cond.Wait()
+		}
+		peer, err := s.store.RandomPeer()
+		s.cond.L.Unlock()
```

Comment:
not necessary to call `RandomPeer` while holding the lock; the store has its own internal mutex. Probably the `SyncerStore` docstring should note that implementations are assumed to be thread-safe. (This is also true of `wallet.Store`.)